### PR TITLE
Move to structured logging

### LINKED ISSUE #222: move discv5 to structured logging
transform logging statements from using formating in strings to structured logging, for example:
```diff
- debug!("Node set to disconnected: {}", node_id)
+ debug!(%node_id, "Node set to disconnected")
```
This would make it easier to make sense of emitted logs

### PR DESCRIPTION
**Pull request by jmezo**
## Description

Changes logging statements from using string formatting to structured logging. Closes #222 

## Change checklist

- [x] Self-review
- ~~[ ] Documentation updates if relevant~~
- [x] Tests if relevant

### CODE REVIEW

File: src/service.rs [hunk 1]
```diff
@@ -1350,13 +1350,10 @@ impl Service {
...
                         self.peers_to_ping.remove(&node_id);
-                        debug!(
-                            "Could not update ENR from pong. Node: {}, reason: {:?}",
-                            node_id, reason
-                        );
+                        debug!(node=%node_id, ?reason, "Could not update ENR from pong",);
                     }
                     update => {
-                        debug!("Updated {:?}", update)
+                        debug!(?update, "Updated")
```
Comment (ackintosh) [inline src/service.rs:1356]:
```suggestion
                        debug!(update_result = ?update, "ENR has been updated based on the pong")
```

File: src/socket/filter/mod.rs [hunk 2]
```diff
@@ -148,8 +148,8 @@ impl Filter {
             .is_some()
         {
             debug!(
-                "Dropped unsolicited packet from banned node_id: {}",
-                node_address
+                node_id = %node_address,
```
Comment (ackintosh) [inline src/socket/filter/mod.rs:151]:
```suggestion
                node = %node_address,
```

**Review by ackintosh — COMMENTED** [review]
Thank you @jmezo, for this PR! I have left some comments.

Additionally, it would be appreciated if you could update the logging statements to add spaces around `=`, similar to the following:
https://github.com/sigp/discv5/blob/ec561f24304f02fab7873fb087cb225bd4ec14fa/src/handler/mod.rs#L656-L661

**Comment by jmezo** [comment]
Thanks for the review and suggestions @ackintosh. I added the spaces around `=` in [4ca08a2](https://github.com/sigp/discv5/pull/258/commits/4ca08a2be21c5b873aee79a3ead4f939e3e545fc).

Let me know if the commits should be squashed, or if there are any other issues.

File: src/service.rs [hunk 3]
```diff
@@ -840,14 +848,11 @@ impl Service {
...
-                                                info!(
-                                                    "Local UDP ip6 socket updated to: {}",
-                                                    new_ip6,
-                                                );
+                                                info!(%new_ip6, "Local UDP ip6 socket updated");
                                                 self.send_event(Event::SocketUpdated(new_ip6));
                                             }
                                             Err(e) => {
-                                                warn!("Failed to update local UDP ip6 socket. ip6: {}, error: {:?}", new_ip6, e);
+                                                warn!(ip6=%new_ip6, error = ?e, "Failed to update local UDP ip6 socket.");
```
Comment (ackintosh) [inline src/service.rs:855]:
```suggestion
                                                warn!(ip6 = %new_ip6, error = ?e, "Failed to update local UDP ip6 socket.");
```

**Comment by ackintosh** [comment]
FYI: The clippy errors will be fixed by https://github.com/sigp/discv5/pull/259

**Review by ackintosh — APPROVED** [review]
Looks good to me.

**Pull request merged**